document-store/src/main/java/org/hypertrace/core/documentstore/expression/type/FromTypeExpression.java

```diff
@@ -0,0 +1,8 @@
+package org.hypertrace.core.documentstore.expression.type;
+
+import org.hypertrace.core.documentstore.parser.FromTypeExpressionVisitor;
+
+/** Interface representing expression used to transform the source table */
```

Comment:
> Interface representing expression used to transform the source table

Not necessarily all the expressions are used for transforming.
Better to mention about joinable, etc. so that future implementors know that a `JoiningExpression` can implement this interface.

Reply:
by transformation I meant any operation that changes the set of rows that you query from. 
Explicitly added the scope for future implementation of this interface.